src/pages/api/superUser.ts

```diff
@@ -0,0 +1,163 @@
+import type { NextApiRequest, NextApiResponse } from "next";
+import { getToken } from 'next-auth/jwt';
+import { prisma } from '../../lib/prisma';
+
+// Post API Inputs.
+export interface PostSuperUserRequestType {
+    role: string;
+    userID: string;
+}
+
+// Post API Output.
+export interface PostSuperUserResponseType {
+    error?: string;
+}
+
+// Put API Inputs.
+export interface PutSuperUserRequestType {
+    role: string;
+    userID: string;
+}
+
+// Put API Outputs.
+export interface PutSuperUserResponseType {
+    error?: string;
+}
+
+// Delete API Inputs.
+export interface DeleteSuperUserRequestType {
+    role: string;
+    tourID?: string;
+    userID?: string;
+}
+
+// Delete API Outputs.
+export interface DeleteSuperUserResponseType {
+    error?: string;
+}
+
+export default async function handler (
+    req: NextApiRequest,
+    res: NextApiResponse
+) {
+    // Updates role to user.
+    if (req.method === "POST") {
+        // Checks JWT token.
+        // const token = await getToken({req});
+        // if (!token)
+        //     return res.status(401).json({ error: "User is not logged in." });
+
+        const { role, userID } = req.body;
+
+        // Error: UserID is not valid.
+        if(typeof userID != "string" || !userID)
+            return res.status(401).json({ error: "UserID is not valid." });
+
+        if (role == "ADMIN") {

```

Comment:
This is dangerous, because as of now, a non admin user can make themselves admin. Only admin users should be allowed to make other admins. You can impliment this by adding the token as so,  `if (token.role === "ADMIN")`